web/src/ApiButton.tsx

```diff
@@ -169,6 +169,9 @@ function ApiButtonWithOptions(props: ApiButtonWithOptionsProps & ButtonProps) {
   const { submit, uiButton, setInputValue, getInputValue, ...buttonProps } =
     props
 
+  let componentType = uiButton.spec?.location?.componentType
+  let tags = { component: componentType }
```

Comment:
wdyt about adding the `annotations` property to the tags? i'm thinking about how we'd want to distinguish between user-defined buttons vs. tilt-defined buttons, like the enable / disable toggle. (this would introduce a dependency between this and #5182, so it'd be fine if you wanted to do that in a separate pr!)

Reply:
i think that's a good idea! but we'd have to be a little careful to make sure we don't accidentally include personally-identifiable data (like if the annotation contains something about the user's project)

Reply:
oh yeah! maybe we could just allowlist the known tilt annotations and ignore everything else.